src/ui/public/agg_types/metrics/top_hit.js

```diff
@@ -0,0 +1,118 @@
+    name: 'top_hits',
+    title: 'Top Hit',
+    makeLabel: function (aggConfig) {
+      const prefix = aggConfig.params.sortOrder.val === 'desc' ? 'Last' : 'First';
+      return `${prefix} ${aggConfig.params.field.displayName}`;
+    },
+    params: [
+      {
+        name: 'field',
+        onlyAggregatable: false,
+        showAnalyzedWarning: false,
+        filterFieldTypes: function (vis, value) {
+          if (vis.type.name === 'table' || vis.type.name === 'metric') {
+            return true;
+          }
+          return value === 'number';
+        },
+        write(agg, output) {
+          const field = agg.params.field;
+          output.params = { size: 1 };
+
+          if (field.scripted) {
+            output.params.script_fields = {
+              [ field.name ]: {
+                script: {
+                  inline: field.script,
+                  lang: field.lang
+                }
+              }
+            };
+          } else {
+            // do not try to get the doc_values for IP fields, since it is
+            // an internal representataion of the IP and so of no use for display.
+            if (field.type !== 'ip' && field.doc_values) {
+              output.params.docvalue_fields = [ field.name ];
+            }
+            output.params._source = field.name;
+          }
+        }
+      },
+      {
+        name: 'sortField',
+        type: 'field',
+        editor: null,
+        filterFieldTypes: [ 'number', 'date', 'ip',  'string' ],
+        default: function (agg) {
+          return agg.vis.indexPattern.timeFieldName;
+        },
+        write: _.noop // prevent default write, it is handled below
+      },
+      {
+        name: 'sortOrder',
+        type: 'optioned',
+        default: 'desc',
+        editor: topSortEditor,
+        options: [
+          { display: 'Descending', val: 'desc' },
+          { display: 'Ascending', val: 'asc' }
+        ],
+        write(agg, output) {
+          const sortField = agg.params.sortField;
+          const sortOrder = agg.params.sortOrder;
+
+          if (sortField.scripted) {
+            output.params.sort = [
+              {
+                _script: {
+                  script: {
+                    inline: sortField.script,
+                    lang: sortField.lang
+                  },
+                  type: sortField.type,
+                  order: sortOrder.val
+                }
+              }
+            ];
+          } else {
+            output.params.sort = [
+              {
+                [ sortField.name ]: {
+                  order: sortOrder.val
+                }
+              }
+            ];
+          }
+        }
+      }
+    ],
+    getValue(agg, bucket) {
+      const hits = _.get(bucket, `${agg.id}.hits.hits`);
+      if (!hits || !hits.length) {
+        return null;
+      }
+      const path = agg.params.field.name;
+      let values = agg.vis.indexPattern.flattenHit(hits[0])[path];
```

Comment:
Actually, it seems that arrays are properly handled by the field format, and it even takes care of custom formatters. Maybe the Discover page shows that warning for another reason ?
I tried using the `formatField` method before but got some issue because of the field format:

- in my test I had the number `3200` which was formatted into the string `3,200` by the `formatField` method
- that gets turned into `3` due to the [`parseFloat`](https://github.com/elastic/kibana/blob/1cddaab2c473103810eb47a016e9ca7290e36f6c/src/ui/public/stringify/types/_numeral.js#L19) method.

I then tried to fix the format of the `top hit` metric to `string` (using the `getFormat` method) but that seemed going too far (and it was not always perfect too...).

For example, with the data below:

```sh
curl -XPOST 'localhost:9200/array/Array' -d '
{
   "foo": [
      {
         "bar": 1
      },
      {
         "bar": 2
      }
   ],
   "num": 1
}
'

curl -XPOST 'localhost:9200/array/Array' -d '
{
   "foo": {
      "bar": [
         2,
         3
      ]
   },
   "num": 2
}
```

And the following field formatter on the `foo.bar` field: `0,0.000`.

I have the following data table when sorting on the `num` field:

- ascending order

![2016-11-29-234734_622x461_scrot](https://cloud.githubusercontent.com/assets/795879/20733958/3a24c9dc-b68e-11e6-8e1c-4a8457ceae61.png)

- descending order

![2016-11-29-234754_575x459_scrot](https://cloud.githubusercontent.com/assets/795879/20733977/52009b44-b68e-11e6-8265-72e6c954f158.png)

For the record, this is the version of ES I am checking this on:

```json
{
  "name" : "k0AsXKW",
  "cluster_name" : "esvm-dev",
  "cluster_uuid" : "cnhHcFz4SmK73pqyh22ZSw",
  "version" : {
    "number" : "6.0.0-alpha1",
    "build_hash" : "32e6fcf",
    "build_date" : "2016-11-23T15:57:31.884Z",
    "build_snapshot" : true,
    "lucene_version" : "6.3.0"
  },
  "tagline" : "You Know, for Search"
}
```